### ISSUE TITLE
Docker version error output going to stdout instead of stderr

### ISSUE THREAD

**Issue by DuckThom** [docker/cli#2777]
**Description**

After updating Docker from CE version 2.3.0.5 to 2.4.0.0 (Client version 19.03.12 to 19.03.13), the error output from `docker version` goes to STDOUT rather than STDERR. This is causing issues with commands to get the server version that rely on stderr output redirection.

**Steps to reproduce the issue:**
1. Install Docker CE 2.4.0.0 but *don't* start it (to trigger the "Cannot connect to daemon" message)
2. Run `docker version -f "{{.Server.Version}}" 2>/dev/null`

**Describe the results you received:**

The output contains the error message, even though the error output is being redirected by `2>/dev/null`

```
➜ docker version -f "{{.Server.Version}}" 2>/dev/null

Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Describe the results you expected:**

Pre-19.03.13 yields the correct result: no error output.

```
➜ docker version -f "{{.Server.Version}}" 2>/dev/null

```

**Output of `docker version` (this time with the server running 😉 ):**

```
➜ docker version
Client: Docker Engine - Community
 Cloud integration  0.1.18
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
➜ docker info
Client:
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 19.03.13
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fba4e9a7d01810a393d5d25a3621dc101981175
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 1.944GiB
 Name: docker-desktop
 ID: PXRQ:VWVG:Y56H:HGYS:M6U6:IKSJ:XG4J:BZ7Y:XGQD:GYND:M4JD:QR5H
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 39
  Goroutines: 46
  System Time: 2020-10-03T13:01:57.376848655Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Running on Mac OS 10.15.7 (19H2)

**Comment by olivierlemasle** [comment]
It works correctly with the latest released Docker version on Ubuntu 20.04, and on `master` branch of this project:

```
$ docker version 2>/dev/null
Client: Docker Engine - Community
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:18:20 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
$ docker version -f "{{.Server.Version}}" 2>error
$ cat error
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Comment by cpuguy83** [comment]
Closing in favor of #2981 since there's more comments on that.
Indeed I think this is related to Docker desktop and the weird stuff being done for integrating https://github.com/docker/compose-cli
